Sources/StreamChat/Controllers/DatabaseObserver/ListChange.swift

```diff
@@ -189,35 +190,54 @@ class ListChangeAggregator<DTO: NSManagedObject, Item>: NSObject, NSFetchedResul
                 log.warning("Skipping the update from DB because `newIndexPath` is missing for `.insert` change.")
                 return
             }
-            currentChanges.append(.insert(item, index: index))
+            currentChanges.append(.insert(dto, index: index))
 
         case .move:
             guard let fromIndex = indexPath, let toIndex = newIndexPath else {
                 log.warning("Skipping the update from DB because `indexPath` or `newIndexPath` are missing for `.move` change.")
                 return
             }
-            currentChanges.append(.move(item, fromIndex: fromIndex, toIndex: toIndex))
+            currentChanges.append(.move(dto, fromIndex: fromIndex, toIndex: toIndex))
 
         case .update:
             guard let index = indexPath else {
                 log.warning("Skipping the update from DB because `indexPath` is missing for `.update` change.")
                 return
             }
-            currentChanges.append(.update(item, index: index))
+            currentChanges.append(.update(dto, index: index))
 
         case .delete:
             guard let index = indexPath else {
                 log.warning("Skipping the update from DB because `indexPath` is missing for `.delete` change.")
                 return
             }
-            currentChanges.append(.remove(item, index: index))
+            currentChanges.append(.remove(dto, index: index))
 
         default:
             break
         }
     }
 
     func controllerDidChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
-        onDidChange?(currentChanges)
+        // Model conversion is safe when all the changes have been processed (Core Data's _processRecentChanges can be called if conversion triggers additional fetch requests).
+        let itemChanges = currentChanges.compactMap { dtoChange in
```

Comment:
This makes sense. Does this potentially fix the current workaround we have the `depth` stuff?

Reply:
Unfortunately no because the circular cycle can still occur with MessageDTO's replies contains a quoted message.